registry/auth.go

```diff
@@ -232,7 +251,8 @@ func ResolveAuthConfig(config *cliconfig.ConfigFile, index *IndexInfo) cliconfig
 		stripped := url
 		if strings.HasPrefix(url, "http://") {
 			stripped = strings.Replace(url, "http://", "", 1)
-		} else if strings.HasPrefix(url, "https://") {
+		}
+		if strings.HasPrefix(url, "https://") {
```

Comment:
> If there's any case where the url is something like `http://https://something` there's something wrong upstream IMO.

Exactly, which is why it should error. With your change it would silently be accepted.